tests/units/feature_extraction/test_feature_calculations.py

```diff
@@ -1221,26 +1222,47 @@ def test_count_above(self):
-        self.assertEqualPandasSeriesWrapper(count_above, [np.NaN, 0, 1] * 3, 2/3, t=0)
-        self.assertEqualPandasSeriesWrapper(count_above, [np.NINF, 0, 1] * 3, 2/3, t=0)
+        self.assertEqualPandasSeriesWrapper(count_above, [0.1, 0.2, 0.3] * 3, 2 / 3, t=0.2)
+        self.assertEqualPandasSeriesWrapper(count_above, [np.NaN, 0, 1] * 3, 2 / 3, t=0)
+        self.assertEqualPandasSeriesWrapper(count_above, [np.NINF, 0, 1] * 3, 2 / 3, t=0)
         self.assertEqualPandasSeriesWrapper(count_above, [np.PINF, 0, 1] * 3, 1, t=0)
         self.assertEqualPandasSeriesWrapper(count_above, [np.NaN, 0, 1] * 3, 0, t=np.NaN)
         self.assertEqualPandasSeriesWrapper(count_above, [np.NINF, 0, np.PINF] * 3, 1, t=np.NINF)
-        self.assertEqualPandasSeriesWrapper(count_above, [np.PINF, 0, 1] * 3, 1/3, t=np.PINF)
+        self.assertEqualPandasSeriesWrapper(count_above, [np.PINF, 0, 1] * 3, 1 / 3, t=np.PINF)
 
     def test_count_below(self):
         self.assertEqualPandasSeriesWrapper(count_below, [1] * 10, 1, t=1)
-        self.assertEqualPandasSeriesWrapper(count_below, list(range(10)), 1/10, t=0)
-        self.assertEqualPandasSeriesWrapper(count_below, list(range(10)), 6/10, t=5)
-        self.assertEqualPandasSeriesWrapper(count_below, [0.1, 0.2, 0.3] * 3, 2/3, t=0.2)
-        self.assertEqualPandasSeriesWrapper(count_below, [np.NaN, 0, 1] * 3, 1/3, t=0)
-        self.assertEqualPandasSeriesWrapper(count_below, [np.NINF, 0, 1] * 3, 2/3, t=0)
-        self.assertEqualPandasSeriesWrapper(count_below, [np.PINF, 0, 1] * 3, 1/3, t=0)
+        self.assertEqualPandasSeriesWrapper(count_below, list(range(10)), 1 / 10, t=0)
+        self.assertEqualPandasSeriesWrapper(count_below, list(range(10)), 6 / 10, t=5)
+        self.assertEqualPandasSeriesWrapper(count_below, [0.1, 0.2, 0.3] * 3, 2 / 3, t=0.2)
+        self.assertEqualPandasSeriesWrapper(count_below, [np.NaN, 0, 1] * 3, 1 / 3, t=0)
+        self.assertEqualPandasSeriesWrapper(count_below, [np.NINF, 0, 1] * 3, 2 / 3, t=0)
+        self.assertEqualPandasSeriesWrapper(count_below, [np.PINF, 0, 1] * 3, 1 / 3, t=0)
         self.assertEqualPandasSeriesWrapper(count_below, [np.NaN, 0, 1] * 3, 0, t=np.NaN)
-        self.assertEqualPandasSeriesWrapper(count_below, [np.NINF, 0, np.PINF] * 3, 1/3, t=np.NINF)
+        self.assertEqualPandasSeriesWrapper(count_below, [np.NINF, 0, np.PINF] * 3, 1 / 3, t=np.NINF)
         self.assertEqualPandasSeriesWrapper(count_below, [np.PINF, 0, 1] * 3, 1, t=np.PINF)
 
+    def test_benford_correlation(self):
+        # A test with list of random values
+        np.random.seed(42)
+        random_list = np.random.uniform(size=100)
+
+        # Fibonacci series is known to match the Newcomb-Benford's Distribution
+        fibonacci_list = [0, 1]
+        for i in range(2, 200):
+            fibonacci_list.append(fibonacci_list[i - 1] + fibonacci_list[i - 2])
+
+        # A list of equally distributed digits (returns NaN)
```

Comment:
Is this expected to return nan? If so, why?

Reply:
The correlation between two variables x and y is:

<img src="https://render.githubusercontent.com/render/math?math=\rho(xy)=\frac{\sigma(xy)}{\sqrt{\sigma_x}\sqrt{\sigma_y}}">

And the covariance between them:

<img src="https://render.githubusercontent.com/render/math?math=\sigma(xy)=\frac{\sum_{i=1}^{n}(x_i-\bar{x})*(y_i-\bar{y})}{n}">

Besides covariance being zero for a series with equal values, <img src="https://render.githubusercontent.com/render/math?math=\sigma_x"> and <img src="https://render.githubusercontent.com/render/math?math=\sigma_y"> are the variances for each series and it will be zero for the series that contains the same values (no variance et all), so the correlation calculus will have a 0/0 (NaN)